packages/firestore/src/api/credentials.ts

```diff
@@ -482,7 +481,15 @@ export class FirebaseAppCheckTokenProvider
             `Error getting App Check token; using placeholder token instead. Error: ${tokenResult.error.message}`
           );
         }
-        return changeListener(tokenResult.token);
+        if (tokenResult.token !== this.latestToken) {
+          logDebug('FirebaseAppCheckTokenProvider', 'Received a new token.');
+          return changeListener(tokenResult.token);
+        }
+        logDebug(
+          'FirebaseAppCheckTokenProvider',
+          'Received a token that is the same as the existing token.'
+        );
```

Comment:
Can we combine the log lines into one? You could just have one return statement, which might the code easier to follow (but that is just personal style).

Reply:
I just want this to be a bit shorter and have less strings, so that our size does not go up by all that much:

```
        const tokenUpdated = tokenResult.token !== this.latestAppCheckToken;
        this.latestAppCheckToken = tokenResult.token;
        logDebug(
          'FirebaseAppCheckTokenProvider',
          `Received ${tokenUpdated ? 'new' : 'existing'} token.`
        );
        return tokenUpdated
          ? changeListener(tokenResult.token)
          : Promise.resolve();
```

Our log lines make up a significant percentage of our SDK size.

Reply:
Our log lines make up a significant percentage of our SDK size.

> Ah! I did not know this. I'll keep that in mind.